nixos/modules/services/x11/window-managers/xmonad.nix

```diff
@@ -83,7 +83,7 @@ in
       session = [{
         name = "xmonad";
         start = if (cfg.config != null) then ''
-          ${xmonadBin}
```

Comment:
I think the more glaring issue is here is we're still starting it in different ways that doesn't exactly make sense.
Why would we not use `systemd-cat` in case of `cfg.config != null`?

How about
```
start = let
  xmonadCommand = if (cfg.config != null) then xmonadBin else "${xmonad}/bin/xmonad";
in ''
   systemd-cat -t xmonad ${xmonadCommand} &
   waitPID=$!
  '';
'';

Reply:
that seems reasonable.

Reply:
Ok pull updated.